src/main/fc/stats.c

```diff
@@ -0,0 +1,42 @@
+
+#include "platform.h"
+
+#ifdef STATS
+
+#include "fc/stats.h"
+
+#include "drivers/time.h"
+#include "navigation/navigation.h"
+
+#include "fc/config.h"
+#include "config/parameter_group.h"
+#include "config/parameter_group_ids.h"
+
+PG_REGISTER_WITH_RESET_FN(statsConfig_t, statsConfig, PG_STATS_CONFIG, 0);
+
+void pgResetFn_statsConfig(statsConfig_t *instance)
+{
+    instance->stats_enabled = 0;
+    instance->stats_total_time = 0;
+    instance->stats_total_dist = 0;
+}
+
+static uint32_t arm_millis;
+static uint32_t arm_distance_cm;
+
+void statsOnArm(void)
+{
+    arm_millis      = millis();
+    arm_distance_cm = getTotalTravelDistance();
+}
+
+void statsOnDisarm(void)
+{
+    if (statsConfig()->stats_enabled) {
```

Comment:
I can do a time check. Can be even more - 10s? 20s? No real flights are such short.
But I wouldn't rely on distance, as for GPS-less machines it will be probably always zero.